### PR TITLE
Disable colors in the output

### PR DESCRIPTION
**Pull request by eunomie**
It's a proposal to implement #205 

Color package has been taken from skaffold as mentionned in the issue. I don't know how you want to handle the copyright notice, if it's needed to be updated or not.

Two things have been changed from the skaffold's version:
- the default color is set to `None`
- a `NoColor` boolean has been introduced and `ColoredOutput` extends `IsTerminal` to know if colors should be printed

Tests has been updated.

Dependencies has been updated to be able to add go-cmp dependency for the tests.

The ~~first~~ third commit of the PR allows to automatically disable colors in the output when the output is not a terminal (like a file).

The ~~second~~ fourth introduces a `--no-color` flag that allows to force to disable colors, even if the output is a terminal. It can help on some CI by example.

In both cases, logrus configuration hasn't been changed, maybe I should do it when `--no-color` is set, WDYT?

### CODE REVIEW

File: pkg/color/formatter.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2019 The Skaffold Authors
```
Comment (nkubala) [inline pkg/color/formatter.go]: this line should look something like `Copyright 2019 Google Inc. All rights reserved.`

File: pkg/color/formatter.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	// NoColor allow to force to not output colors.
+	NoColor = false
+)
+
+// Color can be used to format text using ANSI escape codes so it can be printed to
+// the terminal in color.
+type Color int
+
+var (
+	// LightRed can format text to be displayed to the terminal in light red, using ANSI escape codes.
```
Comment (nkubala) [inline pkg/color/formatter.go]: I think you can remove these comments. it's pretty clear what each of these colors is doing, and is mostly visual clutter :) maybe one comment above the start of the `var` section about how these are all ANSI escape codes and that's it.

File: pkg/color/formatter.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	Yellow = Color(33)
+	// Blue can format text to be displayed to the terminal in blue, using ANSI escape codes.
+	Blue = Color(34)
+	// Purple can format text to be displayed to the terminal in purple, using ANSI escape codes.
+	Purple = Color(35)
+	// Cyan can format text to be displayed to the terminal in cyan, using ANSI escape codes.
+	Cyan = Color(36)
+	// White can format text to be displayed to the terminal in white, using ANSI escape codes.
+	White = Color(37)
+	// None uses ANSI escape codes to reset all formatting.
```
Comment (nkubala) [inline pkg/color/formatter.go]: this comment can stay

File: pkg/color/formatter.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	None = Color(0)
+
+	// Default default output color for output from Skaffold to the user
```
Comment (nkubala) [inline pkg/color/formatter.go]: s/`Skaffold`/`container-structure-test`

File: pkg/color/formatter_test.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2019 The Skaffold Authors
```
Comment (nkubala) [inline pkg/color/formatter_test.go]: update this copyright line as well

File: pkg/color/formatter_test.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+*/
+
+package color
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/testutil"
```
Comment (nkubala) [inline pkg/color/formatter_test.go]: hmm, we don't want to depend on skaffold here. can you copy the relevant methods from the `testutil` package into the container-structure-test codebase? they can live in another `testutil` package here.

File: pkg/color/formatter_test.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+)
+
+func compareText(t *testing.T, expected, actual string, expectedN int, actualN int, err error) {
+	t.Helper()
+	if err != nil {
+		t.Errorf("Did not expect error when formatting text but got %s", err)
```
Comment (nkubala) [inline pkg/color/formatter_test.go]: nit: errors shouldn't be capitalized

File: pkg/color/formatter_test.go [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+		t.Errorf("Formatting not applied to text. Expected \"%s\" but got \"%s\"", expected, actual)
+	}
+}
+
+func TestFprint(t *testing.T) {
+	defer func(f func(io.Writer) bool) { ColoredOutput = f }(ColoredOutput)
+	ColoredOutput = func(io.Writer) bool { return true }
+
+	var b bytes.Buffer
+	n, err := Green.Fprint(&b, "It's not easy being")
```
Comment (nkubala) [inline pkg/color/formatter_test.go]: :frog:

**Comment by eunomie** [comment]
Thanks for the review @nkubala 
I think I've fixed the comments.
I've added one commit to update the dependencies, to be able to add needed dependency for new tests.
And the commit to add skaffold colors has been split to be separated from its usage.

**Comment by eunomie** [comment]
Hi @nkubala 
WDYT about the updated PR? Is that good for you?
Thanks

**Pull request merged**